src/dsp_tools/commands/excel2json/json_header.py

```diff
@@ -0,0 +1,249 @@
+from dsp_tools.commands.excel2json.models.input_error import MissingValuesProblem
+from dsp_tools.commands.excel2json.models.input_error import MoreThanOneRowProblem
+from dsp_tools.commands.excel2json.models.input_error import PositionInExcel
+from dsp_tools.commands.excel2json.models.input_error import Problem
+from dsp_tools.commands.excel2json.models.input_error import RequiredColumnMissingProblem
+from dsp_tools.commands.excel2json.models.json_header import Descriptions
+from dsp_tools.commands.excel2json.models.json_header import EmptyJsonHeader
+from dsp_tools.commands.excel2json.models.json_header import FilledJsonHeader
+from dsp_tools.commands.excel2json.models.json_header import JsonHeader
+from dsp_tools.commands.excel2json.models.json_header import Keywords
+from dsp_tools.commands.excel2json.models.json_header import Prefixes
+from dsp_tools.commands.excel2json.models.json_header import Project
+from dsp_tools.commands.excel2json.models.json_header import User
+from dsp_tools.commands.excel2json.models.json_header import UserRole
+from dsp_tools.commands.excel2json.models.json_header import Users
+from dsp_tools.commands.excel2json.utils import check_contains_required_columns
+from dsp_tools.commands.excel2json.utils import check_required_values_get_position_in_excel
+from dsp_tools.commands.excel2json.utils import read_and_clean_all_sheets
+from dsp_tools.models.exceptions import InputError
+
+
+def get_json_header(excel_filepath: Path) -> JsonHeader:
+    """
+    Returns the header of the JSON file.
+    If an Excel file with the information is provided it is filled out.
+    Otherwise it will return a header with the fields left blank.
+
+    Args:
+        excel_filepath: path to the excel file
+
+    Returns:
+        JsonHeader object
+
+    Raises:
+        InputError: If the file does not conform to the specifics
+    """
+    if not excel_filepath.is_file():
+        print("No json_header.xlsx file found in the directory, empty header was added.")
+        return EmptyJsonHeader()
+    sheets_df_dict = read_and_clean_all_sheets(excel_filepath)
+    sheets_df_dict = {x.lower(): df for x, df in sheets_df_dict.items()}
+    if compliance_problem := _do_formal_compliance(sheets_df_dict):
+        raise InputError(compliance_problem.execute_error_protocol())
+    result = _process_file(sheets_df_dict)
+    if isinstance(result, ExcelFileProblem):
+        raise InputError(result.execute_error_protocol())
+    print("json_header.xlsx file used to construct the json header.")
+    return result
+
+
+def _do_formal_compliance(df_dict: dict[str, pd.DataFrame]) -> ExcelFileProblem | None:
+    expected_sheets = ["prefixes", "project", "description", "keywords"]
+
+    def _check_df(sheet: str) -> str | None:
+        if (df := df_dict.get(sheet)) is None:
+            return sheet
+        if len(df) == 0:
+            return sheet
+        return None
+
+    if empty := [x for x in expected_sheets if _check_df(x)]:
```

Comment:
maybe I'm misremembering - but I think this logic can be achieved with the `any()` function

Reply:
I want to use the empty, so I could do it like this:
`if any(empty := [x for x in expected_sheets if _check_df(x)])`
What do you think would be the advantage?

Reply:
probably there is none here, I think I missed the fact that you're re-using `empty` below (which, by the way is very confusingly named, because this evaluates true when the list is non-empty 😉 )